src/nvim/testdir/setup.vim

```diff
@@ -1,10 +1,61 @@
 " Common preparations for running tests.
 
 " Only load this once.
-if exists('s:did_load')
-  finish
+if 1
+  if exists('s:did_load')
+    finish
+  endif
+  let s:did_load = 1
+endif
+
+" Make sure 'runtimepath' and 'packpath' does not include $HOME.
+set rtp=$VIM/vimfiles,$VIMRUNTIME,$VIM/vimfiles/after
+if has('packages')
+  let &packpath = &rtp
+endif
+
+" Only when the +eval feature is present. 
+if 1
+  " Make sure the .Xauthority file can be found after changing $HOME.
+  if $XAUTHORITY == ''
+    let $XAUTHORITY = $HOME . '/.Xauthority'
+  endif
+
+  " Avoid storing shell history.
+  let $HISTFILE = ""
+
+  " Make sure $HOME does not get read or written.
+  " It must exist, gnome tries to create $HOME/.gnome2
+  let $HOME = getcwd() . '/XfakeHOME'
+  if !isdirectory($HOME)
+    call mkdir($HOME)
+  endif
+endif
+
+" Nvim: we are using "-u NONE" (to disable syntax etc reliably).
+" But keeping unix.vim is good in general (mentioned in help), and required
+" for oldesttest likely still.
+if exists('+shellslash')
+  " Native Windows build.
+  " Use default shell on Windows to avoid segfault, caused by TUI
+  " TODO: revisit?!
```

Comment:
TUI shouldn't segfault anymore but the code that follows should be kept because of `SHELL`.